groups/application/sg.tf

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------------------
+# GFN Security Group and rules
+# ------------------------------------------------------------------------------
+module "gfn_app_ec2_security_group" {
+  source  = "terraform-aws-modules/security-group/aws"
+  version = "~> 3.0"
+
+  name        = "sgr-${var.application}-app-001"
+  description = "Security group for the ${var.application} app ec2"
+  vpc_id      = data.aws_vpc.vpc.id
+
+  computed_ingress_with_source_security_group_id = [
+    {
+      rule                     = "http-8080-tcp"
+      source_security_group_id = module.gfn_app_ec2_security_group.this_security_group_id
```

Comment:
Are you allowing 8080 from the sec group itself?
If so there is a buit in rule for this module called self-self which will be easier.